sim/midas/src/main/cc/simif_emul.h

```diff
@@ -47,23 +49,31 @@ class simif_emul_t : public simif_t {
    * writes directly into the backing memory (0 cycles). See
    * simif_emul_t::load_mems.
    */
-  static mm_t *slave[MEM_NUM_CHANNELS];
+  mm_t **slave;
   /**
    * @brief A model of FPGA-addressable CPU-host memory.
    *
    * In metasimulations, FPGA-managed AXI4 transactions read and write to this
    * AXI4 memory subordinate as a proxy for writing into actual host-CPU DRAM.
    * The driver-side of FPGAManagedStreams inspect circular buffers hosted here.
    */
-  static mm_t *cpu_mem;
+  mm_t *cpu_mem;
 
-private:
+  uint64_t main_time = 0;
+
+  virtual void finish() = 0;
+
+protected:
   // The maximum number of cycles the RTL simulator can advance before
   // switching back to the driver process. +fuzz-host-timings sets this to a
   // value > 1, introducing random delays in axi4 tranactions that MMIO and
   // bridge streams.
   int maximum_host_delay = 1;
-  void advance_target();
+
+  std::string waveform = "dump.vcd";
+  uint64_t memsize = 1L << MEM_ADDR_BITS;
+
+  virtual void advance_target() = 0;
```

Comment:
It be good to explain what this does now that it's going to be implemented by different subclasses.